java/core/lance-jni/src/error.rs

```diff
@@ -97,6 +97,7 @@ impl From<Utf8Error> for Error {
     fn from(source: Utf8Error) -> Self {
         Self::InvalidArgument {
             message: source.to_string(),
+            location: location!(),
```

Comment:
This should use `track_caller` location instead of the conversion site location.

Reply:
see example here: https://github.com/lancedb/lance/blob/main/rust/lance-core/src/error.rs#L154

Reply:
Thanks for the example @chebbyChefNEQ ! Addressed the comments, PTAL